RandomMCPlayer.java

```diff
@@ -168,7 +170,192 @@ private int simPlay(int depthLimit) {
 			return score;
 		}
 	}
+
+	private float getGridUtility() {
+		float gridUtility = 0;
+		for (int row = 0; row < SIZE; row++) {
+			gridUtility += getHandUtility(getHandByRow(row), true);
+		}
+		for (int col = 0; col < SIZE; col++) {
+			gridUtility += getHandUtility(getHandByColumn(col), false);
+		}
+		return gridUtility;
+	}
+
+	private Card[] getHandByRow(int row) {
+		Card[] hand = new Card[SIZE];
+		for (int col = 0; col < SIZE; col++) {
+			hand[col] = grid[row][col];
+		}
+		return hand;
+	}
+
+	private Card[] getHandByColumn(int col) {
+		Card[] hand = new Card[SIZE];
+		for (int row = 0; row < SIZE; row++) {
+			hand[row] = grid[row][col];
+		}
+		return hand;
+	}
+
+	private float getHandUtility(Card[] hand, boolean isRow) {
+		float utility = 0;
+		int numCards = 0;
+		for (int i = 0; i < SIZE ; i++) {
+			if (hand[i] != null) numCards++;
+		}
+		if (numCards == 0) utility = 0;
+		else if (numCards == 5) utility = system.getHandScore(hand);
+		else if (numCards == 4) utility = getExpectedValue(hand);
+		else utility = 3; // temp
```

Comment:
For the `else` case you can just use the `system.getHandScore(hand)` again. Meaning you can use your expected value for 4 cards, and all others (1,2,3,5 cards) can use the system one.
Something like this:
```suggestion
		else if (numCards == 4) utility = getExpectedValue(hand);
		else utility = system.getHandScore(hand);
```